etlman/projects/tests/test_models.py

```diff
@@ -19,3 +22,46 @@ def test_django_model_str_methods(self):
         assert builded_object.name == str(builded_object)
         builded_object = StepFactory.build()
         assert builded_object.name == str(builded_object)
+
+
+@pytest.mark.django_db
+class TestPipelineModel:
+    def test_run_pipeline_single_step(self):
+        """
+        Pipeline.run_pipeline() method creates the correct PipelineRun object in the database.
+        """
+        pipeline = Pipeline(name=PipelineFactory.build().name)
+        pipeline.project = ProjectFactory()
+        pipeline.save()  # Saved pipeline to avoid integrity error due to atomic transaction
+        PipelineScheduleFactory(pipeline=pipeline)
+        StepFactory(pipeline=pipeline)
+        backend = TestBackend()
+        pipeline.run_pipeline(backend=backend)
+        PipelineRun.objects.count() == 1
+        last_run_pipeline = PipelineRun.objects.all().first()
+        assert pipeline.name == last_run_pipeline.pipeline.name
+        assert pipeline.project == last_run_pipeline.pipeline.project
+
+    def test_run_pipeline_multiple_steps(self):
```

Comment:
I'd love to recreate this test. I tried using bulk create but was unsuccessful since we want all Steps to have a one to many relationship to the same Pipeline.

Reply:
What happens if you run `StepFactory()` twice with the same pipeline?

Reply:
It creates them, but with repeated information:
`<QuerySet [<PipelineRun: Elizabeth Houston>, <PipelineRun: Elizabeth Houston>, <PipelineRun: Elizabeth Houston>]>`

Reply:
That's probably okay; it looks like the `__str__` method includes only the pipeline name ATM. You could update that to include something that is unique to the `PipelineRun` if you want to confirm those objects are in fact different (maybe a date, or the PK).